freshmaker/pyxis.py

```diff
@@ -0,0 +1,252 @@
+
+        return self._pagination("operators/indices", request_params)
+
+    def _get_bundles_per_index_image(self, index_images, params=None):
+        """
+        Get bundle images for all index images
+
+        :param list index_images: list of index images
+        :param dict params: parameters to add to GET request
+        :return: bundle images per index image
+        :rtype: dict
+        """
+        if params is None:
+            request_params = {}
+        else:
+            request_params = copy.deepcopy(params)
+        bundles_per_index_image = {}
+        # we need 'bundle_path_digest' to find ContainerImage of that bundle
+        include_fields = ['data.channel_name', 'data.version',
+                          'data.related_images', 'data.bundle_path_digest',
+                          'data.bundle_path']
+        request_params['include'] = ','.join(include_fields)
+        for index_image in index_images:
+            path = index_image.get('path', '')
+            if not path:
+                continue
+            request_params['filter'] = f"source_index_container_path=={path}"
+            bundles_per_index_image[path] = self._pagination('operators/bundles',
+                                                             request_params)
+
+        return bundles_per_index_image
+
+    def get_latest_bundles(self, index_images, params={}):
+        """
+        Get latest bundle images per channel per index image
+
+        :param list index_images: list of index images
+        :param dict params: parameters to add to GET request
+        :return: latest bundle images per channel per index image
+        :rtype: list
+        """
+        bundles_per_index_image = \
+            self._get_bundles_per_index_image(index_images, params)
+
+        ret_bundles = []
+        for index_image, bundles in bundles_per_index_image.items():
+            bundle_per_channel = {}
+            # get latest versions of bundle images per channel
+            for bundle in bundles:
+                channel = bundle.get('channel_name')
+                try:
+                    if channel in bundle_per_channel:
+                        new_ver = bundle.get('version')
+                        old_ver = bundle_per_channel[channel].get('version')
+                        if version.Version(new_ver) > version.Version(old_ver):
+                            bundle_per_channel[channel] = bundle
+                    else:
+                        # Check if the right format of version is used
+                        version.Version(bundle.get('version'))
+                        bundle_per_channel[channel] = bundle
+                # Check if the right format of version is used
+                except version.InvalidVersion as e:
+                    path = bundle.get('bundle_path', 'Unknown bundle path')
+                    log.warning(f"Other format than SemVer is used in "
+                                f"bundle: {path}")
```

Comment:
This is a minor optimization since this will cause the string interpolation to only occur if the logger has the level set to "warning", otherwise, it is just discarded.
```suggestion
                    log.warning("Other format than SemVer is used in bundle: %s", bundle_path)
```